magento/pwa-studio
# [PWA-336] Cart/Checkout Input Errors UI cases

## Description

After Cart/Checkout is done we should make a general pass over the error states to make sure we represent them correctly according to UX desires and the style guide.

Cases to consider:
 * User errors
 * System errors

h5. Open Questions:
 * How should we handle form fields that are out of view of the shopper when the error occurs?
 * Are we going to have both form and field level reporting?
 * Do we have the most common error scenarios / use-cases documented?

Summary of GraphQL Operations that may have error state: https://wiki.corp.magento.com/pages/viewpage.action?pageId=144314396

## Related Issue





* [[PWA-336](https://jira.corp.magento.com/browse/PWA-336)] Cart/Checkout Input Errors UI cases

## Acceptance 



### Verification Stakeholders

### Specification



### Verification Steps


This guy is going to be a little difficult to test, it's quite tricky to simulate errors. You might be able to go offline and use network errors (we've discovered this actually doesn't work in most places), but modifying code to break the queries or mutations is what I did. I'll outline the components touched, and the verification steps are all the same; simulate an error somehow and verify output. I have also included screenshots in the HLD for UX, to make visual review easier.
- CartPage > Product Edit > Loading Options (now renders error)
- CartPage > Remove Item (throws Toast)
- CartPage > Product Edit > Update (renders quantity and option errors now and doesn't close form)
- CartPage > Remove Coupon (now renders error)
- CartPage > Estimate Shipping > Set Address (now renders error)
- CheckoutPage > Select Shipping Method (now renders error and doesn't close if in modal)
- CheckoutPage > Address Book > Apply Address (throws Toast, doesn't close)
- CheckoutPage > Auth/Guest Address > Add/Update Address (renders error, auto scrolls to form top, doesn't close anymore)

## Screenshots / Screen Captures (if appropriate)
See [HLD](https://wiki.corp.magento.com/pages/viewpage.action?pageId=144314396) for tons of screenshots.

## Checklist

* I have updated the documentation accordingly, if necessary.
